### PR TITLE
Remove unsupported arguments from MegatronNMT

### PR DESCRIPTION
**Pull request by MaximumEntropy**
Signed-off-by: MaximumEntropy <sandeep.subramanian.1@umontreal.ca>

# What does this PR do ?

Removes unsupported args from MegatronNMT.

Specifically removes

1. MTDataPreproc (needed for tarred datasets)
2. Tarred and text datasets as options for train_ds.
3. Removes distributed sampler option. Only possible to use megatron batch samplers.
4. Removes associated code in the model like GlobalBatchDataFetcher, train step reconfiguring of MBS etc.

**Collection**: NLP

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by Davood-M — DISMISSED** [review]
looks great, just included some minor optional comments.

File: nemo/collections/nlp/models/machine_translation/megatron_nmt_model.py [hunk 1]
```diff
@@ -454,93 +434,57 @@ def setup_test_data(self, test_data_config: Optional[DictConfig]):
...
         # Parent function will pad microbatches to the same length.
-        return self._process_global_batch_without_megatron_batch_sampler(
-            global_batch, tokenizer=self.encoder_tokenizer
-        )
+        return self._process_global_batch_without_megatron_batch_sampler([batch], tokenizer=self.encoder_tokenizer)
 
     def build_train_valid_test_datasets(self):
         """Builds the train, validation, and test datasets."""
 
         # Builds datasets if the type is tarred or from raw text without memmap.
```
[the remark this answers is not in the document]
Comment (MaximumEntropy) [reply nemo/collections/nlp/models/machine_translation/megatron_nmt_model.py:486]: Good point. Will fix :)

File: nemo/collections/nlp/models/machine_translation/megatron_nmt_model.py [hunk 2]
```diff
@@ -454,93 +434,57 @@ def setup_test_data(self, test_data_config: Optional[DictConfig]):
...
+        )
+        return torch.utils.data.DataLoader(
+            dataset,
+            batch_sampler=batch_sampler,
+            collate_fn=collate_fn,
+            num_workers=cfg.num_workers,
+            pin_memory=cfg.pin_memory,
+        )
 
     def process_global_batch_for_tarred_datasets(self, batch):
```
[the remark this answers is not in the document]
Comment (MaximumEntropy) [reply nemo/collections/nlp/models/machine_translation/megatron_nmt_model.py:467]: Good point. Will fix :)

File: nemo/collections/nlp/models/machine_translation/megatron_nmt_model.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MaximumEntropy) [reply nemo/collections/nlp/models/machine_translation/megatron_nmt_model.py]: Fixed.

File: nemo/collections/nlp/models/machine_translation/megatron_nmt_model.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MaximumEntropy) [reply nemo/collections/nlp/models/machine_translation/megatron_nmt_model.py]: Fixed.

**Pull request merged**